src/components/PhotoSwipe/PhotoSwipe.tsx

```diff
@@ -263,16 +264,17 @@ function PhotoSwipe(props: Iprops) {
     const checkExifAvailable = () => {
         setExif(null);
         setTimeout(() => {
-            const img = document.querySelector(
+            const img: HTMLImageElement = document.querySelector(
                 '.pswp__img:not(.pswp__img--placeholder)'
             );
             if (img) {
-                // @ts-expect-error
-                EXIF.getData(img, function () {
-                    const exif = EXIF.getAllTags(this);
-                    exif.raw = EXIF.pretty(this);
-                    if (exif.raw) {
-                        setExif(exif);
+                exifr.parse(img).then(function (exifData) {
+                    if (!exifData) {
+                        return;
+                    }
+                    exifData.raw = prettyPrintExif(exifData);
+                    if (exifData) {
```

Comment:
Is there a typo here? If not, do we need this if block considering we are already checking for `if (!exifData)`?

Reply:
No, not a typo, we want to set the setExif UI state  only when exifData is not null